### ISSUE TITLE
[Bug]: Dragging the child elements within the group out of the group border will disappear

### ISSUE THREAD

**Issue by more-strive** [fabricjs/fabric.js#9331]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.0-beta12

### In What environments are you experiencing the problem?

Chrome

### Node Version (if applicable)

18.16.0

### Link To Reproduction

https://github.com/dromara/yft-design

### Steps To Reproduce

1.  git clone -b beta12 https://github.com/dromara/yft-design.git
2.  npm install
3.  npm run dev

### Expected Behavior

child elements will not disappear

### Actual Behavior

https://github.com/fabricjs/fabric.js/assets/113762408/73ce8997-69cf-4f1b-95f4-86018f0a3d55

**Comment by asturur** [comment]
Seems a caching clear issue.
From a reproduction point of view we can't really clone the whole app to check.
Can you extract what are you doing with that multi selection? or at least point to the code in the repository where that is handled

**Comment by lh-sn** [comment]
I remember this issue, assumed group rewrite was in progress.

```ts
    const rect1 = new fabric.Rect({
      width: 300,
      height: 300,
      stroke: "#95a5a6",
      top: 300,
      left: 300,
      ...(commonControlConfig as any),
    });

    const rect2 = new fabric.Rect({
      width: 300,
      height: 300,
      stroke: "#95a5a6",
      top: 400,
      left: 400,
      ...(commonControlConfig as any),
    });
    canvas.add(rect1, rect2);

    const group = new fabric.Group([rect1, rect2], {
      interactive: true,
      subTargetCheck: true,
    });
    canvas.add(group);

```
https://codesandbox.io/p/sandbox/fabric-demo-outline-forked-6sg5d9?file=%2Fsrc%2FApp.tsx%3A52%2C14


https://github.com/fabricjs/fabric.js/assets/137523466/e322d8d8-c96a-4c75-86e3-dc2d77982cf6

**Comment by lh-sn** [comment]
Interactive group behavior is confusing. I think only on double click group items should be edited.

**Comment by more-strive** [comment]
![image](https://github.com/fabricjs/fabric.js/assets/113762408/f1e4e99e-d76d-4a4f-9868-e9147e27f65e)
After adding double clicking to enter the group, the sub elements of the group are also grouped and can still be reproduced

https://github.com/fabricjs/fabric.js/assets/113762408/35e52622-d0e4-4c42-a521-5517db0675d4

**Comment by StringKe** [comment]
> I remember this issue, assumed group rewrite was in progress.我记得这个问题，假设组重写正在进行中。
> 
> ```ts
>     const rect1 = new fabric.Rect({
>       width: 300,
>       height: 300,
>       stroke: "#95a5a6",
>       top: 300,
>       left: 300,
>       ...(commonControlConfig as any),
>     });
> 
>     const rect2 = new fabric.Rect({
>       width: 300,
>       height: 300,
>       stroke: "#95a5a6",
>       top: 400,
>       left: 400,
>       ...(commonControlConfig as any),
>     });
>     canvas.add(rect1, rect2);
> 
>     const group = new fabric.Group([rect1, rect2], {
>       interactive: true,
>       subTargetCheck: true,
>     });
>     canvas.add(group);
> ```
> 
> https://codesandbox.io/p/sandbox/fabric-demo-outline-forked-6sg5d9?file=%2Fsrc%2FApp.tsx%3A52%2C14
> 
>  Tab-fabric-group-issue.-.CodeSandbox.mp4

You don't have this problem in your video, but I opened codesandbox and it came up in my browser.

chrome: 119.0.6045.123 arm64
device: Macbook Pro 2021 M1 MAx

**Comment by ShaMan123** [comment]
`objectCaching: false` on the group should solve this for now
We are planning to refactor caching in the future

**Comment by dendrofen** [comment]
> `objectCaching: false` on the group should solve this for now We are planning to refactor caching in the future

not working on 6.0.0-beta16

**Comment by ShaMan123** [comment]
This should be resolved by #9537

**Comment by ShaMan123** [comment]
But not entirely because caching is still the underlying issue.
What will happen from now on is that the layout manager will invalidate it so it will go unnoticed
